### ISSUE TITLE
Update CMake to leverage find_package for hiredis

### ISSUE THREAD

**Issue by prince-chrismc** [sewenew/redis-plus-plus#153]
👋 I like redis and love C++ and I wanted to use both in an upcoming project.

I like using Conan to manage my dependencies so I went and started on https://github.com/conan-io/conan-center-index/pull/4133.

The challenge is the dependency tree is no correctly managed in the `CMakeLists.txt` which can be seen https://github.com/conan-io/conan-center-index/pull/4133#issuecomment-756269622 and it was very well explained https://github.com/conan-io/conan-center-index/pull/4133#discussion_r559258896

> Actually, [it] supports Visual Studio shared, but its CMakeLists.txt doesn't properly propagate dependencies of hiredis (ws2_32 on Windows), leading to undefined symbols when shared hiredis-plus-plus links to static hiredis on Windows.

Being a Conan contributor I racked up some good experience making CMake projects highly available... and I wanted to contribute that back to this project!

***

Would it be possible to take advantage of the CMake targets installed by [`hiredis`](https://github.com/redis/hiredis/blob/297ecbecb71616977ab0bb9b7387d8fa19a5a54f/CMakeLists.txt#L129) and [`hiredis_tls`](https://github.com/redis/hiredis/blob/297ecbecb71616977ab0bb9b7387d8fa19a5a54f/CMakeLists.txt#L205)

This would allow for a simplification in the root CMakeLists.txt

```diff
# hiredis dependency
-find_path(HIREDIS_HEADER hiredis)
-find_library(HIREDIS_LIB hiredis)
+find_package(hiredis CONFIG REQUIRED)
```

Followed by a simple call for

```cmake
target_link_libraries(${STATIC_LIB} PUBLIC hiredis::hiredis)
```

Considering the age of CMake v3.x I think it's safe to drop the CMake 2.x support since it's no longer available on any system by default. However this might be undesirable.

**Comment by SpaceIm** [comment]
Here are some improvements taking advantage of hiredis config file (since 1.0.0): https://github.com/prince-chrismc/conan-center-index/blob/c779e5134e9416bc5d9130b16d158687d8d50482/recipes/redis-plus-plus/all/patches/fix-hiredis-consumption.patch
It allows robust build on all platforms with or without tls. Generally speaking, redis-plus-plus CMakeLists deserves a little bit more love (export targets in a config file and/or create a pkg-config file, allow to build static without PIC).

**Comment by prince-chrismc** [comment]
https://github.com/prince-chrismc/conan-center-index/pull/8/commits/c779e5134e9416bc5d9130b16d158687d8d50482

**Comment by sewenew** [comment]
Hi all, sorry for the inconvenience. I was not familiar with modern cmake, and recently I did some research. Now,  I've already rewritten CMakeLists.txt, and the application code could simply use `find_package` to depend on redis-plus-plus.

Could you please help to review the [CMakeLists.txt](https://github.com/sewenew/redis-plus-plus/blob/master/CMakeLists.txt) and [test/CMakeLists.txt](https://github.com/sewenew/redis-plus-plus/blob/master/test/CMakeLists.txt)? If everything is fine, I'll publish a new release.

Thanks a lot, and sorry for the late reply!

Regards

**Comment by sewenew** [comment]
@SpaceIm 

> export targets in a config file

Done

>  and/or create a pkg-config file

Done

>  allow to build static without PIC

I add `REDIS_PLUS_PLUS_BUILD_STATIC_WITH_PIC` option. If you want to built static lib without PIC, you can set `-DREDIS_PLUS_PLUS_BUILD_STATIC_WITH_PIC=OFF`.

Thanks for your suggestion!

Regards

**Comment by prince-chrismc** [comment]
Looks really good taking a quick glance! 👏 

At the next release I'll try to make time to add it to CCI and if there's an issue I'll come back with a PR!

**Comment by sewenew** [comment]
@prince-chrismc I've tested the latest code with several compilers, and all works well. I've published a new release: [1.2.3](https://github.com/sewenew/redis-plus-plus/releases/tag/1.2.3). 

If you have any problem on redis-plus-plus, feel free to let me know :)

Regards
